Discovery/config_dialog.py

```diff
@@ -26,48 +26,128 @@
         settings.beginGroup("/Discovery")
 
-        # connections
+        # init config list
+        if not settings.value("config_list"):
+            settings.setValue("config_list", [])
+        config_list = settings.value("config_list")
 
-        self.cboConnection.addItem('')
-        for conn in dbutils.get_postgres_connections():
-            self.cboConnection.addItem(conn)
+        # prev version compatibility settings
+        if self.prev_version_config_available():
+            config_list.append("")
+            settings.setValue("config_list", config_list)
+
+        # if empty, add config
+        if not config_list:
+            config_list.append("New config")
+            settings.setValue("config_list", config_list)
+
+        self.cboConnection.addItem("")
+
+        for key in config_list:
+            self.configOptions.addItem(key)
+
+        if self.configOptions.count():
+            self.configOptions.setCurrentIndex(0)
+
+        self.key = self.configOptions.currentText() if self.configOptions.currentIndex() >= 0 else ""
+
+        if not self.configOptions.count():
+            self.enable_form(False)
+        self.set_form_fields(self.key, True)
+        self.chkMarkerTime.stateChanged.connect(self.time_checkbox_changed)
 
-        self.init_combo_from_settings(self.cboConnection, "connection")
+    def prev_version_config_available(self):
+        settings = QSettings()
+        settings.beginGroup("/Discovery")
+
+        conn = settings.value("connection")
+        if conn:
+            return True
+        return False
+
+
+    def validate_nameField(self):
+        settings = QSettings()
+        settings.beginGroup("/Discovery")
+        config_list = settings.value("config_list")
+        key = self.cboName.text()
+
+        if self.validate_key(key, config_list):
+            self.cboName.setStyleSheet("")
+            self.lblMessage.setText("")
+        else:
+            self.lblMessage.setText("<font color=red>Connection name is too short or already exists!</font>")
+            self.cboName.setStyleSheet("QLineEdit {background-color: pink;}")
+
+    # connected to buttonBox.accepted()
+    def validate_and_accept(self):
+        settings = QSettings()
+        settings.beginGroup("/Discovery")
+        config_list = settings.value("config_list")
+        key = self.cboName.text()
+
+        if self.validate_key(key, config_list):
+            self.config_combo.setCurrentIndex(self.configOptions.currentIndex())
+            self.accept()
+        else:
+            self.cboName.setStyleSheet("QLineEdit {background-color: pink;}")
+
+    def set_form_fields(self, key = "", is_init = False):
+
+        QApplication.setOverrideCursor(Qt.WaitCursor)
+        settings = QSettings()
+        settings.beginGroup("/Discovery")
+
+        if key:
+            self.cboName.setText(key)
+        elif is_init:
+            self.cboName.setText("config1")
+        else:
+            self.cboName.setText("")
```

Comment:
For better readability, maybe it would be good to get rid of the is_init argument and the name would be decided by the caller?